src/core/agent/services/connectionService.ts

```diff
@@ -66,44 +67,57 @@ class ConnectionService extends AgentService {
     this.props.eventEmitter.on(EventTypes.ConnectionStateChanged, callback);
   }
 
+  onConnectionAdded() {
+    this.props.eventEmitter.on(
+      EventTypes.ConnectionStateChanged,
+      (event: ConnectionStateChangedEvent) => {
+        if (event.payload.url) {
+          this.resolveOobi(event.payload.url, event.payload.isMultiSigInvite);
+        }
+      }
+    );
+  }
+
+  onConnectionRemoved() {
+    this.props.eventEmitter.on(
+      EventTypes.ConnectionRemoved,
+      (data: ConnectionRemovedEvent) =>
+        this.deleteConnectionById(data.payload.connectionId!)
+    );
+  }
+
   @OnlineOnly
   async connectByOobiUrl(url: string): Promise<OobiScan> {
     const multiSigInvite = url.includes("groupId");
+    const connectionId = new URL(url).pathname
+      .split("/oobi/")
+      .pop()!
+      .split("/")[0];
 
-    // @TODO - foconnor: We shouldn't emit this if it's a multiSigInvite, but the routing will break if we don't.
-    // To fix once we handle errors for the scanner in general.
-    this.props.eventEmitter.emit<ConnectionStateChangedEvent>({
-      type: EventTypes.ConnectionStateChanged,
-      payload: {
-        isMultiSigInvite: multiSigInvite,
-        connectionId: undefined,
-        status: ConnectionStatus.PENDING,
-      },
-    });
+    const alias = new URL(url).searchParams.get("name") ?? new Salter({}).qb64;
 
-    const operation = await this.resolveOobi(url, multiSigInvite);
-    const connectionId =
-      operation.done && operation.response
-        ? operation.response.i
-        : new URL(url).pathname.split("/oobi/").pop()?.split("/")[0];
     const connectionMetadata: any = {
-      alias: operation.alias,
+      alias,
       oobi: url,
-      pending: !operation.done,
+      pending: true,
     };
     const groupId = new URL(url).searchParams.get("groupId") ?? "";
-    const connectionDate = operation.response?.dt ?? new Date();
+    const connectionDate = new Date().toISOString();
     const connection = {
       id: connectionId,
       connectionDate,
-      oobi: operation.metadata.oobi,
-      status: ConnectionStatus.CONFIRMED,
-      label: operation.alias,
+      oobi: url,
+      status: ConnectionStatus.PENDING,
+      label: alias,
       groupId,
     };
 
     if (multiSigInvite) {
+      const operation = await this.resolveOobi(url, multiSigInvite);
+      connectionMetadata.alias = operation.alias;
```

Comment:
We shouldn't need this line. In `const connection = {` above, we should have `alias,` instead of `label: alias`, right?

Reply:
I don't think so. We shouldn't need this line but const `connection = { above, we should have alias, instead of label: alias` seems to be incorrect because the `connection` is receiving type `ConnectionShortDetails` and the `label` is a required field in this type.

Reply:
But `connectionMetadata` is passed to `createConnectionMetadata` to create the record. The record has ALIAS not label. What does `ConnectionShortDetails` have to do with it?